lib/JsonLdDocumentLoader.js

```diff
@@ -18,8 +21,32 @@ class JsonLdDocumentLoader {
     this.documents.set(url, document);
   }
 
+  /**
+   * Adds a DID method specific driver to the cached resolver.
+   *
+   * @param {object} didMethodDriver - A did-io style did method driver, such
+   *   as did-method-key and did-veres-one. (Exposes a `.method` property,
+   *   and a `.get()` function.)
+   */
+  addResolver(didMethodDriver) {
+    this.didDrivers.use(didMethodDriver);
```

Comment:
Hmm, wait, this only works with `didDrivers`? Why? It seems like we don't have the right abstraction here.

Reply:
So here's the thing. There's only two things we ever use documentLoaders for, in all of our libraries. One is loading contexts. And the other is resolving DID documents (and keys in those documents). That's it. I'm sure there's a third use case somewhere, but I have yet to encounter it. So this is not a fringe, complicated use case that we're supporting with this method. It's literally used every time we reach for a documentLoader.

Reply:
Can we involve the URL scheme in the API here somehow? Like, you can set the resolver for `did:` to be that `CachedResolver` thing -- and then you can get it and add things to it, etc?

I'm trying to think ahead for other use cases and for the fact that there are projects that do not involve DIDs that do not want this.

Reply:
@dlongley I'm not sure I fully understand, but I'm open to suggestions.

(If other projects do not want to use DIDs, they don't have to use the `addResolver()` method, right?)

Reply:
> (If other projects do not want to use DIDs, they don't have to use the addResolver() method, right?)

No, there's a did-io import that always happens that they can't avoid. If the API shape were such that you could add a resolver for a given scheme (e.g., `did:`), then you could attach `did-io` only as desired vs. having it always be present even when you don't need it (which is particularly frustrating on web builds).

Reply:
Ok, got it.

> which is particularly frustrating on web builds

`did-io` is actually tiny (just a handful of functions) -- you're probably thinking of `did-cli` or something. But sure, I can adjust the API so that it's not required (and is passed in optionally, externally)

Reply:
Right -- and there may be other schemes that people would want to hook their own resolvers up to (like https).

Reply:
@dlongley - Refactored, added `addProtocolHandler()`, removed `did-io` from dependencies.